fix case insensitive name exists query

dorianjp opened the pull request:
fix, previous name exists query was case sensitive, changed to insensitive

AntoineAugusti reviewed (changes requested):
Could you add a test case proving that the `lower` function is applied?

dorianjp commented:
> Could you add a test case proving that the `lower` function is applied?

@AntoineAugusti good call, I fixed related test case and added another to demonstrate lower works in two scenarios with and without the same service ID.

AntoineAugusti commented on tests/app/service/test_rest.py:
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
+    email_from
+):
+    service = create_service(service_name='unique', email_from='unique')
+    different_service_id = '111aa111-2222-bbbb-aaaa-111111111111'
```
You should be able to get a random UUID here with `uuid.uuid4()` (you can refactor in your other tests if you feel like it)

dorianjp replied:
yes, I just reused the existing hardcoded Id floating around in the tests prior

dorianjp replied:
changed to uuid method

AntoineAugusti commented on tests/app/service/test_rest.py:
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
+    name,
+    email_from
+):
+    service = create_service(service_name='unique', email_from='unique')
```
I think you're not using the `email_from` from your parameters here and you should

dorianjp replied:
removed unused param

AntoineAugusti commented on tests/app/service/test_rest.py:
```diff
@@ -2470,23 +2471,55 @@ def test_is_email_from_unique_returns_200_if_unique(admin_request, notify_db, no
     email_from
 ):
-    create_service(service_name='unique', email_from='unique')
+    service = create_service(service_name='unique', email_from='unique')
```
I think you're not using the `email_from` from your parameters here and you should

dorianjp replied:
removed unused param

dorianjp commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Decouple Email and Service Name](https://trello.com/c/yeyTL5es/158-decouple-email-and-service-name)
